libcxx/test/std/ranges/range.adaptors/range.stride.view/base.pass.cpp

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// constexpr _View base() const& requires copy_constructible<_View>;
+// constexpr _View base() &&;
+
+#include <cassert>
+#include <ranges>
+
+#include "test_iterators.h"
+#include "types.h"
+
+template <typename T>
+constexpr bool hasLValueQualifiedBase(T&& t) {
+  return requires { t.base(); };
+}
+
+constexpr bool test() {
+  int buff[]      = {1, 2, 3, 4, 5, 6, 7, 8};
+  constexpr int N = 8;
+  {
+    using CopyableInputView = CopyableView<cpp17_input_iterator<int*>>;
+    auto str(std::ranges::stride_view<CopyableInputView>(
+        CopyableInputView(cpp17_input_iterator<int*>(buff), cpp17_input_iterator<int*>(buff + N)), 1));
+    assert(*str.base().begin() == *buff);
+    assert(*(std::move(str)).base().begin() == *buff);
+
+    ASSERT_SAME_TYPE(decltype(str.base()), CopyableInputView);
+    ASSERT_SAME_TYPE(decltype(std::move(str).base()), CopyableInputView);
+    static_assert(hasLValueQualifiedBase(str));
```

Comment:
One thing that we started doing since we have concepts is the following pattern:

```c++
{
  std::same_as<CopyableInputView> decltype(auto) result = str.base();
  assert(result.begin() == *buff);
}
{
  // same for rvalue
}
```

You don't have to use this pattern, but it simplifies things a bit.